### PR TITLE
[12.0][IMP] hr_timesheet_sheet: fix amount calculation issues + code simplification

### PR DESCRIPTION
**Pull request by astirpe**
Porting of https://github.com/OCA/hr-timesheet/pull/207 to V12

### CODE REVIEW

File: hr_timesheet_sheet/models/hr_timesheet_sheet.py [hunk 1]
```diff
@@ -56,9 +57,12 @@ def _default_date_end(self):
         return today
 
     def _default_employee(self):
-        emp_ids = self.env['hr.employee'].search(
-            [('user_id', '=', self.env.uid)])
-        return emp_ids and emp_ids[0] or False
+        company = self.env['res.company']._company_default_get()
+        employee = self.env['hr.employee'].search([
```
Comment (alexey-pelykh) [inline hr_timesheet_sheet/models/hr_timesheet_sheet.py]: Сan we go directly to return w/p extra variable? Likewise for company?

File: hr_timesheet_sheet/models/hr_timesheet_sheet.py [hunk 2]
```diff
@@ -407,16 +422,10 @@ def action_timesheet_refuse(self):
...
     def button_add_line(self):
         for rec in self:
-            if rec.state == 'draft':
+            if rec.state in ['new', 'draft']:
                 rec.add_line()
-                rec.reset_add_line()
-        return True
-
-    @api.multi
-    def reset_add_line(self):
```
Comment (alexey-pelykh) [inline hr_timesheet_sheet/models/hr_timesheet_sheet.py]: I'd keep this extension point

**Review by alexey-pelykh — CHANGES_REQUESTED** [review]

**Comment by alexey-pelykh** [comment]
@astirpe many thanks for the porting!

File: hr_timesheet_task_required/tests/test_hr_timesheet_task_required.py [hunk 3]
```diff
@@ -32,31 +32,41 @@ def setUpClass(cls):
...
+            'company_id': cls.user.company_id.id,
+        })
+
     def test_timesheet_line_task_required(self):
         with self.assertRaises(ValidationError):
             self.AnalyticLine.create({
                 'name': 'test',
                 'project_id': self.project_1.id,
                 'unit_amount': 10,
+                'employee_id': self.employee.id,
```
Comment (alexey-pelykh) [inline hr_timesheet_task_required/tests/test_hr_timesheet_task_required.py]: Something in hr_timesheet_sheet triggered this?

Comment (astirpe) [reply hr_timesheet_task_required/tests/test_hr_timesheet_task_required.py]: No. Branch 12.0 seems broken. See PR https://github.com/OCA/hr-timesheet/pull/212, travis fails as well.

The cause is that module `hr_timesheet_employee_required` is installed before module `hr_timesheet_task_required`.

Comment (astirpe) [reply hr_timesheet_task_required/tests/test_hr_timesheet_task_required.py]: I'll create a separate PR if tests become green.

Comment (alexey-pelykh) [reply hr_timesheet_task_required/tests/test_hr_timesheet_task_required.py]: Riiight, my bad - hr_timesheet_employee_required has to be off by default

Comment (alexey-pelykh) [reply hr_timesheet_task_required/tests/test_hr_timesheet_task_required.py]: To avoid side effects

Comment (alexey-pelykh) [reply hr_timesheet_task_required/tests/test_hr_timesheet_task_required.py]: Yet that makes no sense.....

Comment (astirpe) [reply hr_timesheet_task_required/tests/test_hr_timesheet_task_required.py]: Ok, then it could be fixed directly in travis config file. :+1:

Comment (astirpe) [reply hr_timesheet_task_required/tests/test_hr_timesheet_task_required.py]: Opened a dedicated PR https://github.com/OCA/hr-timesheet/pull/213 for the tests of `hr_timesheet_task_required`.

**Comment by astirpe** [comment]
Commit ~~https://github.com/OCA/hr-timesheet/pull/211/commits/3bb749c52b785e86ab2201103eaebf049b012d5c~~ https://github.com/OCA/hr-timesheet/pull/211/commits/8cfea2f25442999ce346a584741f6fa74dbfeb23 makes the matrix lines created with the batch create. This should speeds up the refresh of the summary tab when any onchange is done in the details tab.

File: hr_timesheet_sheet/models/hr_timesheet_sheet.py [hunk 4]
```diff
@@ -259,11 +259,11 @@ def _compute_line_ids(self):
...
                 continue
             matrix = sheet._get_data_matrix()
-            lines = self.env['hr_timesheet.sheet.line']
+            vals_list = []
             for item in sorted(matrix, key=lambda l: self._sort_matrix(l)):
-                vals = sheet._get_default_sheet_line(matrix, item)
-                lines |= self.env['hr_timesheet.sheet.line'].create(vals)
+                vals_list.append(sheet._get_default_sheet_line(matrix, item))
                 sheet.clean_timesheets(matrix[item])
+            lines = self.env['hr_timesheet.sheet.line'].create(vals_list)
```
Comment (alexey-pelykh) [inline hr_timesheet_sheet/models/hr_timesheet_sheet.py]: Do we need `lines`? Maybe just assign to `line_ids`?

Comment (alexey-pelykh) [reply hr_timesheet_sheet/models/hr_timesheet_sheet.py]: And `self.env['hr_timesheet.sheet.line']` may be moved out of the loop

**Review by alexey-pelykh — APPROVED** [review]

**Comment by alexey-pelykh** [comment]
@aheficent what would you say about this?

**Comment by alexey-pelykh** [comment]
@mreficent @pedrobaeza this one is forward-port of recently-merged PR

**Review by MiquelRForgeFlow — APPROVED** [review]

**Comment by alexey-pelykh** [comment]
@pedrobaeza please hit the button 👍

**Comment by pedrobaeza** [comment]
I'm the blind button clicker :yum:

**Comment by alexey-pelykh** [comment]
We guide you! )

**Pull request merged**